app/uploaders/exported_user.rb

```diff
@@ -10,10 +10,13 @@ def store_dir
   end
 
   def extension_allowlist
-    %w[gz]
+    %w[gz zip json]
   end
 
   def filename
-    "#{model.username}_diaspora_data_#{secure_token}.json.gz" if original_filename.present?
+    if original_filename.present? # rubocop:disable Style/GuardClause
+      extension = File.extname(@filename) if @filename
+      "#{model.username}_data_#{secure_token}#{extension}"
```

Comment:
Its for down- and uploads. A user also might upload an uncompressed json. 
This will rescue the multi-point suffice
```suggestion
    return if original_filename.blank?

    filename_parts = original_filename.split(".")
    filename_parts.slice!(0)
    extensions = filename_parts.join(".")
    "#{model.username}_data_#{secure_token}.#{extensions}"
```